### PR TITLE
Do not cache index.html to prevent issues where Dashboard doesn't load when the ray version is upgraded.

### PR DESCRIPTION
**Pull request by alanwguo**
## Why are these changes needed?
- Do not cache index.html to prevent issues where Dashboard doesn't load when the ray version is upgraded.
- Explictly sets a long cache time for static files to improve performance for files we know will not change

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by architkulkarni — APPROVED** [review]
Looks good to me. Assuming there's dashboard tests in CI which cover this codepath.

File: dashboard/http_server_head.py [hunk 1]
```diff
@@ -109,11 +109,13 @@ async def get_index(self, req) -> aiohttp.web.FileResponse:
                 "This error message is harmless and can be ignored. "
                 f"Error: {e}"
             )
-        return aiohttp.web.FileResponse(
+        resp = aiohttp.web.FileResponse(
             os.path.join(
                 os.path.dirname(os.path.abspath(__file__)), "client/build/index.html"
             )
         )
+        resp.headers["Cache-Control"] = "no-cache"
```
Comment (architkulkarni) [inline dashboard/http_server_head.py:117]: Is there any risk of the key `"Cache-Control"` not being present?

Comment (alanwguo) [reply dashboard/http_server_head.py:117]: nope, the default behavior is determined by browser when it's not set.

**Comment by alanwguo** [comment]
Please help me merge as well @architkulkarni

**Pull request merged**